userspace/falco/yaml_helper.h

```diff
@@ -77,42 +78,77 @@ class yaml_helper
 		get_node(node, key);
 		if(node.IsDefined())
 		{
-			std::string value = node.as<std::string>();
+			auto value = node.as<std::string>();
 
 			// Helper function to convert string to the desired type T
 			auto convert_str_to_t = [&default_value](const std::string& str) -> T {
+				if (str.empty())
+				{
+					return default_value;
+				}
+
+				if constexpr (std::is_same_v<T, std::string>)
+				{
+					return str;
+				}
 				std::stringstream ss(str);
 				T result;
-				if (ss >> result) return result;
+				if (ss >> std::boolalpha >> result) return result;
```

Comment:
Enforce `std::boolalpha` so that `true` and `false` are correctly parsed to booleans.